src/lib.rs

```diff
@@ -410,6 +410,15 @@ impl<R: io::Read> DecodeReaderBytes<R, Vec<u8>> {
     pub fn new(rdr: R) -> DecodeReaderBytes<R, Vec<u8>> {
         DecodeReaderBytesBuilder::new().build(rdr)
     }
+
+    /// Return the encoding used by this decoder, which may have been
+    /// detected automatically or set manually.
+    ///
+    /// If automatic encoding detection is being used, the return value
+    /// of this method can change during the life of the decoder.
```

Comment:
This seems very non-ideal to me. How can one correctly use this method?

Reply:
The same way you would use it with `encoding_rs` https://docs.rs/encoding_rs/latest/encoding_rs/struct.Decoder.html#method.encoding

It's really only possible on the first read rather than at any random point, I can clarify that in the docs.  But this is partially why I prefer having a public `rdr.detect_encoding()` or  `rdr.detect_encoding_with(...)?;` method which would make the check explicit.

Reply:
I'm not convinced that's correct. Who is to say that the first read pulls in enough bytes for encoding detection to work? It is perfectly legal and correct for read calls to write at most one byte into the caller provided buffer.

Reply:
I believe that is the problem that [`utils::read_full`](https://github.com/BurntSushi/encoding_rs_io/blob/master/src/util.rs#L222-L243) exists to solve, do you think it would be sufficient to just use this function to fill the first buffer?

Reply:
I would like to see a real example how of this method is being used, and ideally, it should be ported to a documentation test. I don't understand what this is supposed to be used for.